coremltools/models/model.py

```diff
@@ -334,7 +334,11 @@ def predict(self, data, useCPUOnly=False, **kwargs):
 
             try:
                 from ..libcoremlpython import _MLModelProxy
+            except Exception as e:
+                print("exception loading model proxy: %s\n" % e)
+                _MLModelProxy = None
             except:
+                print("exception while loading model proxy.\n")
```

Comment:
I don't think this code block will ever be executed. I think `except Exception as e:` caughts everything.

Reply:
While deriving exceptions from `Exception` is recommended Python practice, I don't know of anything in Python which *guarantees* that all exceptions are derived from `Exception`.  This prints "poor exception", for example:

```python
#!python
class Nothing: pass
try:
  raise Nothing
except Exception as e:
  print("exception: ", e)
except:
  print("poor exception")
```

Reply:
https://docs.python.org/3/library/exceptions.html.